dashboard/lib/services/lti.rb

```diff
@@ -192,6 +192,16 @@ def self.sync_section_roster(lti_integration, lti_section, nrps_section)
           nrps_member: nrps_member
         )
         user_was_new = user.new_record?
+        # Update name if different from the NRPS response
+        unless user_was_new
+          nrps_member_message = Policies::Lti.issuer_accepts_resource_link?(issuer) ? nrps_member[:message].first : nrps_member
+          if user.teacher?
+            user.name = Services::Lti.get_claim_from_list(nrps_member_message, Policies::Lti::TEACHER_NAME_KEYS)
+          else
+            user.name = Services::Lti.get_claim_from_list(nrps_member_message, Policies::Lti::STUDENT_NAME_KEYS)
+            user.family_name = Services::Lti.get_claim(nrps_member_message, :family_name)
+          end
```

Comment:
Can we move this section to a separate method and reuse it here?
https://github.com/code-dot-org/code-dot-org/blob/ed56c8a1da0b32ea65ef053d9ec99bd9d63dcbcf/dashboard/lib/services/lti.rb#L16-L23